<launch>

	<!-- VIDEO SOURCE -->
	<arg name="input" default="csi://0"/>
	<arg name="input_width" default="0"/>
	<arg name="input_height" default="0"/>
	<arg name="input_codec" default="unknown"/>
	<arg name="input_loop" default="0"/>

	<include file="$(find ros_deep_learning)/launch/video_source.ros1.launch">
		<arg name="input" value="$(arg input)"/>
		<arg name="input_width" value="$(arg input_width)"/>
		<arg name="input_height" value="$(arg input_height)"/>
		<arg name="input_codec" value="$(arg input_codec)"/>
		<arg name="input_loop" value="$(arg input_loop)"/>
	</include>

	<!-- DETECTNET -->
	<arg name="model_name" default="ssd-mobilenet-v2"/>
	<arg name="model_path" default=""/>
	<arg name="prototxt_path" default=""/>
	<arg name="class_labels_path" default=""/>
	<arg name="input_blob" default=""/>
	<arg name="output_cvg" default=""/>
	<arg name="output_bbox" default=""/>
	<arg name="overlay_flags" default="box,labels,conf"/>
	<arg name="mean_pixel_value" default="0.0"/>
	<arg name="threshold" default="0.5"/>

	<node pkg="ros_deep_learning" type="detectnet" name="detectnet" output="screen">
		<remap from="/detectnet/image_in" to="/video_source/raw"/>
		<param name="model_name" value="$(arg model_name)"/>
		<param name="model_path" value="$(arg model_path)"/>
		<param name="prototxt_path" value="$(arg prototxt_path)"/>
		<param name="class_labels_path" value="$(arg class_labels_path)"/>
		<param name="input_blob" value="$(arg input_blob)"/>
		<param name="output_cvg" value="$(arg output_cvg)"/>
		<param name="output_bbox" value="$(arg output_bbox)"/>
		<param name="overlay_flags" value="$(arg overlay_flags)"/>
		<param name="mean_pixel_value" value="$(arg mean_pixel_value)"/>
		<param name="threshold" value="$(arg threshold)"/>
	</node>

	<!-- VIDEO OUTPUT -->
	<arg name="output" default="display://0"/>
	<arg name="output_codec" default="unknown"/>
	<arg name="output_bitrate" default="0"/>

	<include file="$(find ros_deep_learning)/launch/video_output.ros1.launch">
		<arg name="topic" value="/detectnet/overlay"/>
		<arg name="output" value="$(arg output)"/>
		<arg name="output_codec" value="$(arg output_codec)"/>
		<arg name="output_bitrate" value="$(arg output_bitrate)"/>
	</include>

</launch>
